lib/ansible/plugins/connection/ssh.py

```diff
@@ -558,6 +528,24 @@ def wrapped(self: Connection, *args: P.args, **kwargs: P.kwargs) -> tuple[int, b
     return wrapped
 
 
+def _clean_shm(func):
+    def inner(self, *args, **kwargs):
+        try:
+            ret = func(self, *args, **kwargs)
+        finally:
+            with contextlib.suppress(AttributeError):
```

Comment:
I had to re-read a few times to remember. But the first could be switched to `if self.shm:` instead of catching an exception. I've done that, because it's actually faster than handling an exception for the common case.

The 2nd needs to say, but they cannot be combined.  We still need to run the code within `if not _HAS_RESOURCE_TRACK:` regardless of the `FileNotFoundError` but only when `self.shm` is set.